### PR TITLE
feat(ingest/powerbi): fix subTypes and add workspace_type_filter

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

File: metadata-ingestion/docs/sources/powerbi/powerbi_pre.md [hunk 1]
```diff
@@ -20,7 +20,8 @@
 | `Report`              | `Dashboard`         |
 | `Page`                | `Chart`             |
 
-If Tile is created from report then Chart.externalUrl is set to Report.webUrl.
+- If `Tile` is created from report then `Chart.externalUrl` is set to Report.webUrl.
+- The `Page` information is unavailable for PowerBI PaginatedReport.
```
Comment (hsheth2) [inline metadata-ingestion/docs/sources/powerbi/powerbi_pre.md:24]: i have no idea what this sentence means? like what is the "Page information"?

File: metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py [hunk 2]
```diff
@@ -69,8 +69,8 @@ class BIAssetSubTypes(StrEnum):
     LOOKER_LOOK = "Look"
 
     # PowerBI
-    POWERBI_TILE = "PowerBI Tile"
-    POWERBI_PAGE = "PowerBI Page"
+    POWERBI_TILE = "Tile"
+    POWERBI_PAGE = "Page"
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/common/subtypes.py:73]: we can keep powerbi in these

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/config.py [hunk 3]
```diff
@@ -445,6 +456,16 @@ class PowerBiDashboardSourceConfig(
...
     )
 
+    workspace_type_filter: List[
+        Literal[
+            "Workspace", "PersonalGroup", "Personal", "AdminWorkspace", "AdminInsights"
+        ]
+    ] = pydantic.Field(
+        default=["Workspace"],
+        description="Ingest the metadata of the workspace where the workspace type corresponds to the specified workspace_type_filter."
+        " Note: This field works in conjunction with 'workspace_id_pattern' and both must be considered when filtering workspaces.",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/config.py:466]:
```suggestion
        " Note: This field works in conjunction with 'workspace_id_pattern'. Both must be matched for a workspace to be processed.",
```

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py [hunk 4]
```diff
@@ -712,15 +698,13 @@ def chart_custom_properties(dashboard: powerbi_data_classes.Dashboard) -> dict:
         info_mcp = self.new_mcp(
             entity_type=Constant.DASHBOARD,
             entity_urn=dashboard_urn,
-            aspect_name=Constant.DASHBOARD_INFO,
             aspect=dashboard_info_cls,
         )
 
         # removed status mcp
         removed_status_mcp = self.new_mcp(
             entity_type=Constant.DASHBOARD,
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py:706]: we can also remove entity_type - that is auto-inferred

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py [hunk 5]
```diff
@@ -41,6 +41,7 @@ class DatasetKey(ContainerKey):
 class Workspace:
     id: str
     name: str
+    type: str
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py:44]: is this also being used as the subtype? if so, we add that as a comment to make it clear

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py [hunk 6]
```diff
@@ -827,7 +808,7 @@ def generate_container_for_workspace(
         container_work_units = gen_containers(
             container_key=self.workspace_key,
             name=workspace.name,
-            sub_types=[BIContainerSubTypes.POWERBI_WORKSPACE],
+            sub_types=[workspace.type],
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py:811]: is BIContainerSubTypes.POWERBI_WORKSPACE used anywhere now?

File: metadata-ingestion/docs/sources/powerbi/powerbi_pre.md (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/docs/sources/powerbi/powerbi_pre.md:24]: removed the information

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py (hunk 6 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py:811]: removed

File: metadata-ingestion/docs/sources/powerbi/powerbi_pre.md [hunk 7]
```diff
@@ -21,7 +21,7 @@
 | `Page`                | `Chart`             |
 
 - If `Tile` is created from report then `Chart.externalUrl` is set to Report.webUrl.
-- The `Page` information is unavailable for PowerBI PaginatedReport.
+- The `Page` is unavailable for PowerBI PaginatedReport.
```
Comment (hsheth2) [inline metadata-ingestion/docs/sources/powerbi/powerbi_pre.md:24]: this is still confusing - let's add PaginatedReport to the concept mapping table which will clean things up

also, how is Report.webUrl mapped to a chart if the report is mapped to a dashboard?

**Review by hsheth2 — APPROVED** [review]
We can address this final comment in a follow-up

File: metadata-ingestion/docs/sources/powerbi/powerbi_pre.md (hunk 7 above)
Comment (sid-acryl) [reply metadata-ingestion/docs/sources/powerbi/powerbi_pre.md:25]: The tile contains visualizations that can be created using a dataset, a text box, or by pinning a report. If a report is pinned within the tile, the `externalUrl` will point to that report.

Comment (hsheth2) [reply metadata-ingestion/docs/sources/powerbi/powerbi_pre.md:25]: I see

**Comment by hsheth2** [comment]
@sid-acryl tests are failing

**Comment by dlahyani** [comment]
@sid-acryl  while this PR adds a nice feature, the drop of filtering objects at the API level has some serious downsides. While previously listing of workspaces took like 2 minutes  for large organization with ~50K workspaces, it now takes forever since the get_groups API returns a lot more objects (>350K in our case) which most of them are not workspaces and we don't care about. The Get Groups as Admin API has a limit of 15 req/m or 50 req/h, thus just listing these 350K objects takes >7 hours.

I suggest to re-introduce the filtering at the API level. If you agree I can open a PR that performs the filtering at the API level while keeping the ability to fetch other types of objects as well.

**Pull request merged**